public/app/features/explore/ErrorContainer.test.tsx

```diff
@@ -1,35 +1,46 @@
 import React from 'react';
-import { DataQueryError } from '@grafana/data';
-import { shallow } from 'enzyme';
-import { ErrorContainer } from './ErrorContainer';
+import { render, screen } from '@testing-library/react';
+import { ErrorContainer, ErrorContainerProps } from './ErrorContainer';
 
-const makeError = (propOverrides?: DataQueryError): DataQueryError => {
-  const queryError: DataQueryError = {
-    data: {
-      message: 'Error data message',
-      error: 'Error data content',
-    },
-    message: 'Error message',
-    status: 'Error status',
-    statusText: 'Error status text',
-    refId: 'A',
-  };
-  Object.assign(queryError, propOverrides);
-  return queryError;
-};
-
-const setup = (propOverrides?: object) => {
-  const props = {
-    queryError: makeError(propOverrides),
-  };
+describe('ErrorContainer', () => {
+  it('should render component and show message', () => {
+    const props: ErrorContainerProps = {
+      queryError: {
+        data: {
+          message: 'Error data message',
+          error: 'Error data content',
+        },
+        message: 'Error message',
+        status: 'Error status',
+        statusText: 'Error status text',
+        refId: 'A',
+      },
+    };
+    render(<ErrorContainer {...props} />);
+    expect(screen.getByText('Query error')).toBeInTheDocument();
+    expect(screen.getByText('Error message')).toBeInTheDocument();
+  });
 
-  const wrapper = shallow(<ErrorContainer {...props} />);
-  return wrapper;
-};
+  it('should render component and show message if message is in data only', () => {
+    const props: ErrorContainerProps = {
+      queryError: {
+        data: {
+          message: 'Error data message',
+          error: 'Error data content',
+        },
+        status: 'Error status',
+        statusText: 'Error status text',
+        refId: 'A',
+      },
+    };
+    render(<ErrorContainer {...props} />);
+    expect(screen.getByText('Query error')).toBeInTheDocument();
+    expect(screen.getByText('Error data message')).toBeInTheDocument();
+  });
 
-describe('ErrorContainer', () => {
-  it('should render component', () => {
-    const wrapper = setup();
-    expect(wrapper).toMatchSnapshot();
+  it('should show unknown error if prop is not passed in', () => {
+    const props: ErrorContainerProps = {};
+    render(<ErrorContainer {...props} />);
+    expect(screen.getByText('Unknown error')).toBeInTheDocument();
```

Comment:
```suggestion

    const alertEl = screen.getByRole('alert');
    expect(alertEl).toBeInTheDocument();
    expect(alertEl).toHaveTextContent('Unknown error');
```

Reply:
Good call on these, and what I didn't realize is the alert is hidden if it is unknown, so this test is even more accurate. Thanks!